whelktool/scripts/cleanups/2019/11/marc-fiction-cleanup/fiction-cleanup.groovy

```diff
@@ -72,26 +71,27 @@ selectBySqlWhere(query, silent: false) { data ->
     }
 }
 
-private boolean hasOnlyHClassifications(work) {
+private boolean everySabClassifcationIsH(work) {
     def classif = work.classification
     classif = classif instanceof Map ? [classif] : classif
-    return classif?.every { c -> hasClassificationH(c) }
+    return classif?.findAll { c -> isSAB(c) }?.every { c -> hasClassificationH(c) }
 }
 
-boolean hasClassificationH(classification) {
+boolean isSAB(classification) {
     def type = classification.'@type' as String
-    def code = classification?.code instanceof String ? [classification.code] : classification.code
 
     if (type == "Classification") {
         def inSchemeCode = classification.inScheme?.code as String
-        if (inSchemeCode && inSchemeCode == "kssb" && classification.inScheme?.'@type' == "ConceptScheme") {
-            return code && code.every { c -> c.startsWith("H") || c.startsWith("uH") }
-        }
+        return inSchemeCode && inSchemeCode == "kssb" && classification.inScheme?.'@type' == "ConceptScheme"
```

Comment:
It can be better to use `isInstanceOf(object, typeName)` instead of direct type value equality comparison (since that checks for subclasses as well).

Reply:
Cool!